server/server.ts

```diff
@@ -0,0 +1,45 @@
+require('dotenv').config();
+import { ApolloServer } from 'apollo-server';
+import { connectDB } from "./mongoConnection";
+
+import { config } from './config';
+
+import { Request, RequestDocument } from './models/requestModel';
+import RequestDataSource from './datasources/requestsDataSource'
+import { RequestsCache } from "./cache";
+
+// TODO: need to make script to build(compile) prod server and to run prod server
+
+
+// //-----------------------------------------------------------------------------
+// // APOLLO SETUP
+// //-----------------------------------------------------------------------------
+
+import { typeDefs } from "./schema";
+import { resolvers } from "./resolvers";
+
+//-----------------------------------------------------------------------------
+// MONGODB CONNECTION AND DATA SOURCES FOR APOLLO
+//-----------------------------------------------------------------------------
+
+// connect to MongoDB and setup data sources
+connectDB(() => {
+    if (config.caching) {
+        RequestsCache.init();
+    }
+});
+
+
+//-----------------------------------------------------------------------------
+// SERVER LAUNCH
+//-----------------------------------------------------------------------------
+
+const server = new ApolloServer({ typeDefs, resolvers, dataSources: () => ({
+    requests: new RequestDataSource()
+}) });
+
+const port = config.port | 4000;
```

Comment:
Isn't the single pipe a bitwise operator? Also shouldn't we be specifying the port as an environmental variable? Since when we deploy it the port comes from the environment.

Reply:
Bitwise OR operator with undefined/null/false-y/0 values to return '4000'. I removed it now since it would not be possible to fail to load the config.ts (it is possible that Node fails to load .env).

I moved all non-secrets to config.ts since .env is in .gitignore and we may be storing all secrets in .env. We can move this around when we determine where our secrets will go.

Reply:
These config values should be shared through the repo, so they need a file to be in that isn't ignored.